### PR TITLE
Add density

### PR DESCRIPTION
**Pull request by tvdeyen**
Adds `density` instance methods to `Cuboid` and `Package`.

### CODE REVIEW

File: lib/physical/cuboid.rb [hunk 1]
```diff
@@ -19,6 +19,11 @@ def volume
       Measured::Volume(dimensions.map { |d| d.convert_to(:cm).value }.reduce(1, &:*), :ml)
     end
 
+    def density
+      return Float::INFINITY if volume.value.zero?
```
Comment (mamhoff) [inline lib/physical/cuboid.rb]: What if any dimension is infinite, as is the case for the default containers? I think in that case the density should be zero, as 
```
d = weight / width * height * INFINITY = weight / INFINITY => 0
```

File: lib/physical/package.rb [hunk 2]
```diff
@@ -37,5 +37,10 @@ def void_fill_weight
 
       Measured::Weight(void_fill_density.value * remaining_volume.value, void_fill_density.unit)
     end
+
+    def density
+      return Float::INFINITY if container.volume.value.zero?
```
Comment (mamhoff) [inline lib/physical/package.rb]: Same concern about infinite container volumes.

File: lib/physical/cuboid.rb (hunk 1 above)
Comment (tvdeyen) [reply lib/physical/cuboid.rb]: Ah, sure. You are right. Will add this. Thanks.

File: lib/physical/package.rb (hunk 2 above)
Comment (tvdeyen) [reply lib/physical/package.rb]: Absolutely. Thanks

File: lib/physical/cuboid.rb (hunk 1 above)
Comment (tvdeyen) [reply lib/physical/cuboid.rb]: Done. I was not able to produce an infinite large `Item`. I added the spec to the `Box` instead.

File: lib/physical/package.rb (hunk 2 above)
Comment (tvdeyen) [reply lib/physical/package.rb]: Done

File: lib/physical/cuboid.rb [hunk 3]
```diff
@@ -19,6 +19,12 @@ def volume
...
+      return 0.0 if volume.value.infinite?
+      weight.convert_to(:g).value / volume.convert_to(:ml).value
```
Comment (mamhoff) [inline lib/physical/cuboid.rb]: Now that I see the `0.0` here, another thought comes to mind: We don't have a unit for the density - although it should really be at least the denominator part of the `g/ml` unit. What about this: 

```rb
Measured::Weight(weight.convert_to(:g).value / volume.convert_to(:ml).value, :g)
```

Another option would be to add a new `Measured::Density` type as detailed in https://github.com/Shopify/measured#namespaces - that would also help for the `void_fill_density` parameter to the `Physical::Package`. 

```rb
Measured::Density(weight.convert_to(:g).value / volume.convert_to(:ml).value, :g_per_ml)
```

Comment (tvdeyen) [reply lib/physical/cuboid.rb]: Thought about the same but decided to not use a unit here on purpose. Adding weight as unit felt wrong to me. Was not aware of the ability to add own units to `Measured`. Let me add the `Measured::Density` in this PR as I think this is the right thing to do now. Great find!

Comment (tvdeyen) [reply lib/physical/cuboid.rb]: Done https://github.com/mamhoff/physical/pull/19/commits/c423d03866335fa58d045dcfef39033a0c560053

Please double check my math in the unit conversion :smile:

File: lib/measured/density.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    :kilograms_per_cubic_meter,
+    :kilograms_per_cubic_metre
+  ]
+
+  unit :g_ml, aliases: [
+    :gram_per_milliliter,
+    :gram_per_millilitre,
+    :grams_per_milliliter,
+    :grams_per_millilitre
+  ], value: '1000 kg_m3'
```
Comment (mamhoff) [inline lib/measured/density.rb]: I think it has to be the other way round, 1 g/ml is 0.001 kg_m3. 

See 
grafik

File: lib/measured/density.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  # UPS Freight classes use pound per cubic feet as density unit
+  unit :lb_ft3, aliases: [
+    :lbs_ft3,
+    :pound_per_cubiq_foot,
+    :pound_per_cubiq_feet,
+    :pounds_per_cubiq_foot,
+    :pounds_per_cubiq_feet
+  ], value: '160.184897305 kg_m3'
```
Comment (mamhoff) [inline lib/measured/density.rb]: grafik

File: lib/measured/density.rb (hunk 4 above)
Comment (tvdeyen) [reply lib/measured/density.rb]: I found the conversion rules API of Measured a bit confusing.
But as I understand it the conversion rule says:

"1 g/ml equals 1000 kg/m3"

What should be what Google tells as as well, right?

File: lib/measured/density.rb (hunk 5 above)
Comment (tvdeyen) [reply lib/measured/density.rb]: Thanks. Will fix this. Where did you find the Google conversion tool? I was only ablt ot find the Volume one. Not the Mass / Volume one. Could you paste the link in here, please?

Comment (tvdeyen) [reply lib/measured/density.rb]: Fixed

Comment (mamhoff) [reply lib/measured/density.rb]: I used https://www.google.com/search?q=1+lb%2Fft3+in+kg%2Fm3 and friends

Comment (tvdeyen) [reply lib/measured/density.rb]: Thanks. You might have found a hidden feature :)

File: lib/measured/density.rb (hunk 4 above)
Comment (mamhoff) [reply lib/measured/density.rb]: Yep, you're right! Thanks for being attentive!

**Comment by tvdeyen** [comment]
@mamhoff 

> One last bit is to do in a subsequent PR: Convert the `void_fill_density` parameter to the `Physical::Package` intializer to use our new `Measured::Density`. After that's done I'll release a new version. Do you want to have a go at it or should I?

Could you do it, please?

**Pull request merged**